app/gui2/src/util/autoBlur.ts

```diff
@@ -2,24 +2,25 @@ import { useEvent } from '@/composables/events'
 import type { Ref } from 'vue'
 
 export function useAutoBlur(root: Ref<HTMLElement | SVGElement | MathMLElement | undefined>) {
-  useEvent(
-    window,
-    'pointerdown',
-    (event) => {
-      if (
-        !root.value?.contains(document.activeElement) ||
-        !(event.target instanceof Element) ||
-        root.value.contains(event.target)
-      )
-        return
-      if (
-        !(document.activeElement instanceof HTMLElement) &&
-        !(document.activeElement instanceof SVGElement) &&
-        !(document.activeElement instanceof MathMLElement)
-      )
-        return
-      document.activeElement.blur()
-    },
-    { capture: true },
+  useEvent(window, 'pointerdown', (event) => blurIfNecessary(root, event), { capture: true })
+}
+
+export function blurIfNecessary(
+  root: Ref<HTMLElement | SVGElement | MathMLElement | undefined>,
+  event: MouseEvent,
+): boolean {
```

Comment:
Please add short docs to these functions. It's not obvious from the name what they do.

Reply:
Added